cmd/cert.go

```diff
@@ -236,6 +236,21 @@ func rotateRKECertificates(ctx context.Context, kubeCluster *cluster.Cluster, fl
 		return nil, fmt.Errorf("Failed to rotate certificates: can't find old certificates")
 	}
 	currentCluster.RotateCertificates = kubeCluster.RotateCertificates
+	if !kubeCluster.RotateCertificates.CACertificates {
+		caCertPKI, ok := rkeFullState.CurrentState.CertificatesBundle[pki.CACertName]
+		if !ok {
+			return nil, fmt.Errorf("[rotateRKECertificates] certificate [%s] not found", pki.CACertName)
+		}
+		caCert := caCertPKI.Certificate
+		if caCert == nil {
+			return nil, fmt.Errorf("[rotateRKECertificates] certificate [%s] is nil", pki.CACertName)
+		}
+		certPool := x509.NewCertPool()
+		certPool.AddCert(caCert)
+		if _, err := caCert.Verify(x509.VerifyOptions{Roots: certPool, KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}}); err != nil {
+			return nil, fmt.Errorf("[rotateRKECertificates] failed to rotate certs: %v", err)
+		}
```

Comment:
nit: could probably put this in common and reuse it at least, gives us a reusable function for something we might need to do elsewhere. Just worth considering but not necessary.

Reply:
I was considering that earlier, but since from `rotateRKECertificates` we'd like to return separate error messages for when cert is nil vs when Verify returns an error for it, it didn't make sense to put this in a separate function, and then determine the type of error that function has returned.